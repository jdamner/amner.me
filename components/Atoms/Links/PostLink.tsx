"use client";

import React from "react";

/* Components */
import Link from "next/link";
import Image from "next/image";

/* Types */
import type { PostLinkType } from "../../../types";

/**
 * Post Link
 *
 * @param {Post} post
 * @return JSX.Element
 */
export default function PostLink({
  thumbnail,
  title,
  slug,
}: PostLinkType): React.JSX.Element {
  const classes = [
    "group",
    "block",
    "bg-white",
    "dark:bg-slate-900",
    "border-2",
    "border-slate-500",
  ];

  const linkClasses = [
    "bg-slate-900",
    "group-hover:bg-slate-700",
    "px-2",
    "text-lg",
    "leading-loose",
    "text-center text-slate-100",
  ];

  return (
    <Link href={slug} className={classes.join(" ")}>
      <Image
        src={thumbnail}
        alt={""}
        width={375}
        height={375}
        className="w-full brightness-90 dark:brightness:75 group-hover:brightness-105"
      />
      <h2 className={linkClasses.join(" ")}>{title}</h2>
    </Link>
  );
}
