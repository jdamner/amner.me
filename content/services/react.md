---
title: Javascript & React
---

I've worked with [React](https://reactjs.org) and [React Native](https://reactnative.dev) for a number of years and find the stateful nature of the React framework to really work well in my head. I love the component based approach and the ability to create reusable components that can be used in multiple places.

This site is build using React and the [NextJS](https://www.nextjs.org/) framework. I parse [markdown](https://daringfireball.net/projects/markdown/) files for the content. This site is totally open source too, so if you're curious about how it's built, you can check out the [source code](https://www.github.com/jdamner/amner.me).

Also, as a statically generated site, it's super fast and easy to deploy.
