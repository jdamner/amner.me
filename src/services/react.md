---
title: React
thumbnail: ../assets/react-logo.png
---

Working with SQL in raw queries, and using frameworks such as Eloquent and WP_Query.
Coaching and supporting a team of developers grow their skills and meet their objectives
Supporting Project Managers and Business Development teams for technical guidance on quoting and supporting project lifecycles.
Debugging and maintaining PHP projects built in many frameworks, including bespoke PHP frameworks developed in house.
Developing systems that interact with complex data, such as data provided via third-party tools and services that run on device and output in raw formats.