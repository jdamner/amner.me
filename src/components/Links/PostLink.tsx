import Link from "next/link";
import Image from "next/image";
import type { post } from "../../types/post.type";
import { useState } from "react";
import Spinner from "../Global/Spinner";
import { useRouter } from "next/router";
import { event } from "../../api/Insights";

export default function PostLink({ post }: { post: post }): JSX.Element {
	const [ isClicked , setIsClicked ] = useState(false);
	const router = useRouter();

	const handleClick = (e: React.MouseEvent<HTMLAnchorElement>) => {
		event('click', { event: e.currentTarget.innerText });
		setIsClicked(true);
		router.events.on('routeChangeComplete', () => {
			setIsClicked(false);
		});
	}

	const classes = [
		'group',
		'mb-3',
		'block',
		'rounded-md',
		'overflow-hidden',
		'w-full',
		'bg-white',
		'relative',
	]

	const linkClasses = [
        'bg-gradient-to-r',
        'from-teal-900',
        'to-emerald-600',
        'group-hover:from-emerald-700',
        'group-hover:to-teal-500',
        'shadow-slate-900',
        'px-4',
        'text-lg',
        'font-sans',
        'font-semibold',
        'tracking-tight',
        'leading-loose',
        'text-center',
        'text-white',
        'normal-case',
    ];


	return (
		<Link href={post.slug} className={ classes.join( ' ' ) } onClick={handleClick}>
			{ isClicked ? <div className="absolute top-0 left-0 w-full h-full bg-white/10 backdrop-blur-sm z-10 rounded-md">
				<Spinner className='flex items-center justify-center w-full h-full'/>
			</div> : null}
			<Image src={post.thumbnail} alt={post.title} loading='lazy' width={300} height={300} className="w-full brightness-90 group-hover:brightness-105 transition" />
			<h2 className={linkClasses.join(' ')}>{post.title}</h2>
		</Link>
	)
}