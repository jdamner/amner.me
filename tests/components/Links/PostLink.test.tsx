import { render } from "@testing-library/react"
import PostLink from "../../../src/components/Links/PostLink"
import { getTestPost } from "../../helper"

describe("PostLink", () => {
  it("renders correctly", () => {
    const { container } = render(<PostLink post={ getTestPost() }/>)
    expect(container).toMatchSnapshot()
  })
})
