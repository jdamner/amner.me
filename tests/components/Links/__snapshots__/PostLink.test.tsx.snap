// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostLink renders correctly 1`] = `
<div>
  <a
    class="group mb-3 block rounded-md overflow-hidden w-full bg-white relative"
    href="test-post"
  >
    <img
      alt="Test Post"
      class="w-full brightness-90 group-hover:brightness-105 transition"
      data-nimg="1"
      decoding="async"
      height="300"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.png&w=640&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.png&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.png&w=640&q=75 2x"
      style="color: transparent;"
      width="300"
    />
    <h2
      class="bg-gradient-to-r from-teal-900 to-emerald-600 group-hover:from-emerald-700 group-hover:to-teal-500 shadow-slate-900 px-4 text-lg font-sans font-semibold tracking-tight leading-loose text-center text-white normal-case"
    >
      Test Post
    </h2>
  </a>
</div>
`;
